libc/string_cpy.c

```diff
@@ -0,0 +1,1269 @@
+	TEST_ASSERT_EQUAL_INT(num, numDest);
+
+	TEST_ASSERT_EQUAL_PTR(&fltDest, memcpy(&fltDest, &flt, sizeof(flt)));
+	TEST_ASSERT_NOT_EMPTY(&fltDest);
+	TEST_ASSERT_FLOAT_IS_DETERMINATE(fltDest);
+	TEST_ASSERT_EQUAL_FLOAT(flt, fltDest);
+
+	TEST_ASSERT_EQUAL_PTR(&ptrDest, memcpy(&ptrDest, &ptr, sizeof(ptr)));
+	TEST_ASSERT_NOT_EMPTY(&ptrDest);
+	TEST_ASSERT_EQUAL_INT(ptr, ptrDest);
+
+	TEST_ASSERT_EQUAL_PTR(&sizDest, memcpy(&sizDest, &size, sizeof(size)));
+	TEST_ASSERT_NOT_EMPTY(&sizDest);
+	TEST_ASSERT_EQUAL_DOUBLE(size, sizDest);
+}
+
+
+TEST(string_memcpy, adjacent)
+{
+	char memStr[BIG_NUMB] = { 0 },
+		 testStr[] = "TEST_STRING",
+		 expVal[BIG_NUMB];
+
+	/* Creating expected value of connected tests strings */
+	strcpy(expVal, testStr);
+	strcat(expVal, testStr);
+
+	/* Copying string in same space but couple addresses onto left to merge text into one combining two similar texts */
+	sprintf(&memStr[BIG_NUMB / 2], "%s", testStr);
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(testStr, &memStr[BIG_NUMB / 2], sizeof(testStr));
+	TEST_ASSERT_EQUAL_PTR(&memStr[(BIG_NUMB / 2) - 11], memcpy(&memStr[(BIG_NUMB / 2) - 11], &memStr[BIG_NUMB / 2], sizeof(testStr) - 1));
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(&memStr[(BIG_NUMB / 2) - 11], expVal, sizeof(testStr) * 2 - 2);
+
+	/* Adding another one iteration of test string */
+	strcat(expVal, testStr);
+
+	/* Same as before but move onto right */
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(testStr, &memStr[BIG_NUMB / 2], sizeof(testStr));
+	TEST_ASSERT_EQUAL_PTR(&memStr[(BIG_NUMB / 2) + 11], memcpy(&memStr[(BIG_NUMB / 2) + 11], &memStr[BIG_NUMB / 2], sizeof(testStr) - 1));
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(&memStr[(BIG_NUMB / 2) - 11], expVal, sizeof(testStr) * 3 - 3);
+}
+
+
+TEST(string_memcpy, one_byte)
+{
+	char testArray[MAX_ASCII_SET] = { 0 },
+		 input[MAX_ASCII_SET] = { 1 };
+	int i;
+
+	TEST_ASSERT_NOT_EQUAL_INT(0, strcmp(input, testArray));
+
+	/* This loop will copy only one byte in a place where the loop iterator pointing this place */
+	for (i = 0; i < MAX_ASCII_SET; i++) {
+		TEST_ASSERT_EQUAL_PTR(&testArray[i], memcpy(&testArray[i], &input[i], 1));
+	}
+
+	/* Checking if all elements were correctly copied */
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(input, testArray, MAX_ASCII_SET);
+}
+
+
+TEST(string_memcpy, nothing)
+{
+	char testArray[MAX_STR_LEN] = TEST_STR1,
+		 input[MAX_STR_LEN] = { 0 },
+		 strText[] = "Lorem ipsum \0 dolor";
+
+	/* Clearing array with support array filled with 0 */
+	TEST_ASSERT_EQUAL_PTR(testArray, memcpy(testArray, input, MAX_STR_LEN));
+	TEST_ASSERT_NOT_EQUAL_INT(0, strcmp(testArray, TEST_STR1));
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(input, testArray, MAX_STR_LEN);
+
+
+	/* Checking no sensitivity for NULL */
+	memset(input, 0, sizeof(input));
+	memcpy(input, strText, sizeof(strText));
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(strText, input, sizeof(strText));
+}
+
+
+TEST(string_memcpy, big)
+{
+	char buff[1024] = { 0 };
+
+	/* Checking capability of handling big blocks of data */
+	TEST_ASSERT_EQUAL_INT(1, big_string());
```

Comment:
You could call it once for each test group, for example in memccpy tests it's called a few times.